### PR TITLE
Fix back-redo-edit action sync

### PR DESCRIPTION
**Pull request by jojorne**
This fix the data loss and the back button.

Known issues:
- [ ] Empty actions are in the list and do not skip
- [x] UI 2 forces skip empty action text and it gets stuck (workaround: use redo of UI 1)

### CODE REVIEW

**Comment by jojorne** [comment]
Working on skipping zero length text...

File: aiserver.py [hunk 1]
```diff
@@ -8675,10 +8672,11 @@ def UI_2_back(data):
 def UI_2_redo(data):
     if koboldai_vars.aibusy:
         return
-    if len(koboldai_vars.actions.get_current_options()) == 1:
-        koboldai_vars.actions.use_option(0)
+    if koboldai_vars.debug:
+        print("redo")
```
Comment (one-some) [inline aiserver.py]: Probably better to use `logger.debug` here

Comment (jojorne) [reply aiserver.py]: it's something UI2 does. it's everywhere 😅

Comment (one-some) [reply aiserver.py]: Probably some old legacy code, i'll have to remember to convert that :P
either way for new code logger should probably be used, it gives more debug info and is prettier. plus if we dont use it, whats the point of having it?

**Comment by ebolam** [comment]
The second checkbox appears to be fixed in PR#328

**Comment by ebolam** [comment]
Correction second checkbox is fixed in PR #333

**Comment by henk717** [comment]
Going to close this because of the editor rewrite that hopefully fixes these things. Thanks for the contribution so far!

**Pull request closed without merging**